pkg/cloud/services/ec2/network.go

```diff
@@ -26,33 +29,53 @@ func (s *Service) ReconcileNetwork() (err error) {
 
 	// VPC.
 	if err := s.reconcileVPC(); err != nil {
+		conditions.MarkFalse(s.scope.AWSCluster, infrav1.VpcReadyCondition, infrav1.VpcReconciliationFailedReason, clusterv1.ConditionSeverityError, err.Error())
 		return err
 	}
+	conditions.MarkTrue(s.scope.AWSCluster, infrav1.VpcReadyCondition)
```

Comment:
Wondering if we can move this call inside the `reconcileVPC` func, just before the final `return nil` call
Similar consideration applies for the following `MarkTrue` calls

Reply:
yes they can. They were left there because I was originally trying to contain condition updates to the controller but now they're all over the services package